homeassistant/components/hive.py

```diff
@@ -0,0 +1,707 @@
+"""Hive Integration - Platform."""
+import logging
+from datetime import datetime
+from homeassistant.helpers.discovery import load_platform
+
+REQUIREMENTS = ['pyhiveapi==0.1.1']
+
+HGO = None
+_LOGGER = logging.getLogger(__name__)
+DOMAIN = 'hive'
+
+
+class HivePlatformData:
+    """Initiate Hive PlatformData Class."""
+
+    min_max_data = {}
+
+
+class HiveSession:
+    """Initiate Hive Session Class."""
+
+    username = ""
+    password = ""
+    data = HivePlatformData()
+    logging = False
+    hass = None
+    core = None
+    heating = None
+    hotwater = None
+    light = None
+    sensor = None
+    switch = None
+
+
+def setup(hass, config):
+    """Setup the Hive platform."""

```

Comment:
This docstring will not pass validation with PEP257.

Reply:
docstring updated as advised